service/src/main/java/com/theokanning/openai/service/AzureOpenAiService.java

```diff
@@ -0,0 +1,224 @@
+        request.setStream(true);
+
+        return stream(api.createCompletionStream(request, deployment, apiVersion), CompletionChunk.class);
+    }
+
+    public ChatCompletionResult createChatCompletion(ChatCompletionRequest request, String apiVersion) {
+        return execute(api.createChatCompletion(request, deployment, apiVersion));
+    }
+
+    public Flowable<ChatCompletionChunk> streamChatCompletion(ChatCompletionRequest request, String apiVersion) {
+        request.setStream(true);
+
+        return stream(api.createChatCompletionStream(request, deployment, apiVersion), ChatCompletionChunk.class);
+    }
+
+    public EmbeddingResult createEmbeddings(EmbeddingRequest request, String apiVersion) {
+        return execute(api.createEmbeddings(request, deployment, apiVersion));
+    }
+
+    public List<File> listFiles(String apiVersion) {
+        return execute(api.listFiles(apiVersion)).data;
+    }
+
+    public File uploadFile(String purpose, String filepath, String apiVersion) {
+        java.io.File file = new java.io.File(filepath);
+        RequestBody purposeBody = RequestBody.create(okhttp3.MultipartBody.FORM, purpose);
+        RequestBody fileBody = RequestBody.create(MediaType.parse("text"), file);
+        MultipartBody.Part body = MultipartBody.Part.createFormData("file", filepath, fileBody);
+
+        return execute(api.uploadFile(purposeBody, body, apiVersion));
+    }
+
+    public File retrieveFile(String fileId, String apiVersion) {
+        return execute(api.retrieveFile(fileId, apiVersion));
+    }
+
+    public FineTuneResult createFineTune(FineTuneRequest request, String apiVersion) {
+        return execute(api.createFineTune(request, apiVersion));
+    }
+
+    public FineTuneResult retrieveFineTune(String fineTuneId, String apiVersion) {
+        return execute(api.retrieveFineTune(fineTuneId, apiVersion));
+    }
+
+    /**
+     * Calls the Open AI api, returns the response, and parses error messages if the request fails
+     */
+    public static <T> T execute(Single<T> apiCall) {
+        try {
+            return apiCall.blockingGet();
+        } catch (HttpException e) {
+            e.printStackTrace();
+            try {
+                if (e.response() == null || e.response().errorBody() == null) {
+                    throw e;
+                }
+                String errorBody = e.response().errorBody().string();
+
+                OpenAiError error = mapper.readValue(errorBody, OpenAiError.class);
+                throw new OpenAiHttpException(error, e, e.code());
+            } catch (IOException ex) {
+                // couldn't parse OpenAI error
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Calls the Open AI api and returns a Flowable of SSE for streaming
+     * omitting the last message.
+     *
+     * @param apiCall The api call
+     */
+    public static Flowable<SSE> stream(retrofit2.Call<ResponseBody> apiCall) {
```

Comment:
```suggestion
    public static Flowable<SSE> stream(Call<ResponseBody> apiCall) {
```
Simplify it to `Call`, as `retrofit2.Call` is already imported.